softlayer/resource_softlayer_vlan.go

```diff
@@ -196,17 +201,27 @@ func resourceSoftLayerVlanRead(d *schema.ResourceData, meta interface{}) error {
 
 	// Subnets
 	subnets := make([]map[string]interface{}, 0)
+	primarySubnets := make([]map[string]interface{}, 0)
 
 	for _, elem := range vlan.Subnets {
 		subnet := make(map[string]interface{})
+		primarySubnet := make(map[string]interface{})
+		validPrimaryType := regexp.MustCompile(`.*PRIMARY.*`)
```

Comment:
To avoid compiling this on each iteration, this should be done in the global scope.